tests/kafkatest/services/verifiable_producer.py

```diff
@@ -68,29 +68,33 @@ def __init__(self, context, num_nodes, kafka, topic, max_messages=-1, throughput
         self.message_validator = message_validator
         self.compression_types = compression_types
 
+        self.security_config = self.kafka.security_config.client_config()
+
         for node in self.nodes:
             node.version = version
         self.acked_values = []
         self.not_acked_values = []
         self.produced_count = {}
-        self.prop_file = ""
+
+    def prop_file(self, node):
+        idx = self.idx(node)
+        prop_file = str(self.security_config)
+        if self.compression_types is not None:
+            compression_index = (idx - 1) % len(self.compression_types)
```

Comment:
Sorry I missed this in the first pass, but it looks like this modulo operation will break in general

e.g. by default, `self.compression_types` is `None`, but `len(None)` raises a TypeError

Also, if compression_types is not None, it seems better to enforce that `len(compression_types) == self.num_nodes` in the constructor.  Otherwise you also have to handle the case where `len(self.compression_types) == 0` since `x % 0 ` raises `ZeroDivisionError`. Do we actually need to do this tricky modulo operation?